page-object-code-gen/src/main/java/com/kaspresso/components/pageobjectcodegen/CreatePageObjectFromUiDump.kt

```diff
@@ -0,0 +1,133 @@
+package com.kaspresso.components.pageobjectcodegen
+
+import com.kaspresso.components.pageobjectcodegen.SupportedViews.collectableElements
+import com.kaspresso.components.pageobjectcodegen.SupportedViews.elementsWithChild
+import org.w3c.dom.Node
+import java.io.File
+import java.nio.charset.Charset
+import javax.xml.parsers.DocumentBuilderFactory
+
+/**
+ * inputs:
+ * 1. Path to xml file with UI Dump
+ * 2. Name of generated class
+ * 3. Path for generated file
+ * output:
+ * Kotlin file with screen code in the same directory as jar execute
+ */
+fun main(vararg args: String) {
```

Comment:
Please, write more documentation (comments, methods documentation or even wiki file like we have in the docs/ folder). Some areas of your code are hard to understand. It would be easier to maintain your code if you express the concept behind code generation